### PR TITLE
Add clear analysis action

### LINKED ISSUE #804: Add Analysis tab action to clear displayed analysis
## Problem

The Analysis tab lets users apply analysis overlays, but there is no obvious way to clear/remove analysis from the map when they decide they do not want any analysis displayed.

This makes analysis feel like a one-way visual state: once an overlay is applied, users may have to manually clean up map layers or restart/reload the workflow to get back to a plain activity map.

## Proposal

Add a clear/reset analysis action in the Analysis tab.

Possible behavior:

- Provide a visible **Clear analysis** / **Remove analysis from map** action near the analysis controls.
- Remove or hide qfit-created analysis layers/overlays from the QGIS map.
- Reset any analysis status/summary copy to a neutral “No analysis displayed” state.
- Preserve the underlying imported/synced activity data and normal map layers.

## Acceptance criteria

- Users can clear analysis overlays from the Analysis tab without deleting activities or other map layers.
- The Analysis tab communicates that no analysis is currently displayed after clearing.
- Clearing is safe to run repeatedly when no analysis is present.
- Tests cover the analysis clear action and state update.

### PR DESCRIPTION
**Pull request by ebelo**
## Summary
- Add a visible **Clear analysis** action to the Analysis tab content.
- Wire the action through the wizard/local-first callback adapters to remove qfit analysis overlays.
- Refresh the dock state after clearing and show neutral “No analysis displayed” copy.

Closes #804.

## Tests
- `python3 -m pytest tests/test_analysis_page.py tests/test_local_first_composition.py tests/test_wizard_shell_composition.py tests/test_qfit_dockwidget_analysis_pure.py -q`
- `python3 -m pytest tests/ -x -q --tb=short`

### CODE REVIEW

File: ui/dockwidget/analysis_page.py [hunk 1]
```diff
@@ -46,16 +47,19 @@ class AnalysisPageState:
         "Optional: calculate heatmaps, corridors, and start points from loaded data."
     )
     input_summary_text: str = "No loaded activity layers available for analysis"
-    result_summary_text: str = "Analysis outputs will appear in the project once generated"
+    result_summary_text: str = "No analysis displayed"
     primary_action_label: str = "Run analysis"
     primary_action_enabled: bool | None = None
     primary_action_blocked_tooltip: str = "Load activity layers before running analysis."
+    clear_action_label: str = "Clear analysis"
+    clear_action_enabled: bool = True
```
[the remark this answers is not in the document]
Comment (ebelo) [reply ui/dockwidget/analysis_page.py:55]: Addressed in `458fb71`: `_clear_analysis_layer()` now only marks the atlas export stale when it actually removes a current/stale qfit analysis layer, and it returns `False` as a true no-op when no analysis is displayed. I also added a regression test covering the empty-clear case so repeated Clear analysis clicks cannot invalidate an atlas without an analysis overlay.
